server/routerlicious/packages/lambdas/src/deli/lambdaFactory.ts

```diff
@@ -86,11 +86,38 @@ export class DeliLambdaFactory extends EventEmitter implements IPartitionLambdaF
         let document: IDocument;
 
         try {
-            gitManager = await this.tenantManager.getTenantGitManager(tenantId, documentId);
-
             // Lookup the last sequence number stored
             // TODO - is this storage specific to the orderer in place? Or can I generalize the output context?
             document = await this.collection.findOne({ documentId, tenantId });
+
+            // Check if the document was deleted prior.
+            if (!isDocumentValid(document)) {
+                // (Old, from tanviraumi:) Temporary guard against failure until we figure out what causing this to trigger.
+                // Document sessions can be joined (via Alfred) after a document is functionally deleted.
+                const errorMessage = `Received attempt to connect to a missing/deleted document.`;
+                context.log?.error(
+                    errorMessage,
+                    { messageMetaData },
+                );
+                Lumberjack.error(errorMessage, getLumberBaseProperties(documentId, tenantId));
+                return new NoOpLambda(context);
+            }
+
+            if (!isDocumentSessionValid(document, this.serviceConfiguration)) {
+                // Session for this document is either nonexistent or exists in a different location.
+                const errMsg =
+                    `Received attempt to connect to invalid session: ${JSON.stringify(document.session)}`;
+                context.log?.error(errMsg, { messageMetaData });
+                Lumberjack.error(errMsg, getLumberBaseProperties(documentId, tenantId));
+                if (this.serviceConfiguration.enforceDiscoveryFlow) {
```

Comment:
If this is false, the flow would still go to the gitmanager call on line 120 below -- just for my understanding, is that intentional, and if yes, wouldnt that gitManager call fail if session is invalid?

Reply:
The subsequent flow would all pass if the load if normal. The gitManager call will fail only in the cases where there is a spike in docs getting recreated like it happened during our deployments. This flag is currently set to true in one of our clusters and the plan is to turn it on everywhere soon.

Also, without this PR, even if we set the flag correctly, we will end up making the riddler calls which is what we are trying to fix.